### PR TITLE
[MIG][12.0] sale_report_delivered_subtotal: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT20186
@chienandalu @Yajo Can you review?

### CODE REVIEW

File: sale_report_delivered_subtotal/reports/sale_report.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class SaleReport(models.Model):
+    _inherit = "sale.report"
+
+    price_subtotal_delivered = fields.Float(
+        string='Total Price Delivered',
+    )
+
+    def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
```
Comment (pedrobaeza) [inline sale_report_delivered_subtotal/reports/sale_report.py]: For avoiding the linter (which is correct that can be dangerous), but with this change it doesn't affect anything:
```suggestion
    def _query(self, with_clause='', fields=None, groupby='', from_clause=''):
```

File: sale_report_delivered_subtotal/reports/sale_report.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        fields = dict(fields or {})
```
Comment (pedrobaeza) [inline sale_report_delivered_subtotal/reports/sale_report.py]: Adapted to the above code:
```suggestion
        if fields is None:
            fields = {}
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You can also squash administrative commits with the first non-administrative one:

Selección_027

File: sale_report_delivered_subtotal/reports/sale_report.py (hunk 2 above)
Comment (yajo) [reply sale_report_delivered_subtotal/reports/sale_report.py]: Actually it would work as it is.

Comment (pedrobaeza) [reply sale_report_delivered_subtotal/reports/sale_report.py]: It's for not creating a new one if not needed and then build a new dict for it, but it's not critical indeed.

**Comment by sergio-teruel** [comment]
Changes done!!

**Review by pedrobaeza — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**